backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/DemoAuthenticationConfiguration.java

```diff
@@ -0,0 +1,153 @@
+            String userName = authHeader.substring(DEMO_AUTHORIZATION_FLAG.length());
+            securityContext.setAuthentication(
+                new TestingAuthenticationToken(userName, null, List.of()));
+          }
+          chain.doFilter(request, response);
+        };
+    return new FilterRegistrationBean<>(filter);
+  }
+
+  @Bean
+  public AuthorizationService getDemoAuthorizationService(DemoUserConfiguration config) {
+    return new DemoUserAuthorizationService(config);
+  }
+
+  @Bean
+  public IdentitySupplier getDemoIdentitySupplier(DemoUserConfiguration config) {
+    return new DemoUserIdentitySupplier(config);
+  }
+
+  /**
+   * Creates and returns a {@link Supplier} that examines the current security context, finds the
+   * username of the current user, and returns the {@link DemoUser} that matches that username. If a
+   * default user has been configured, that user will be returned in the event that there is no
+   * username or the username does not match any known DemoUser.
+   *
+   * @param config the configured list of DemoUsers, with or without a default user set.
+   */
+  private static Supplier<Optional<DemoUser>> getCurrentDemoUserSupplier(
+      DemoUserConfiguration config) {
+    return () -> {
+      Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+      String userName = null;
+      if (auth == null) { // have to allow defaulting to work even when there is no authentication
+        LOG.warn("No authentication found: current user will be default (if available)");
+      } else if (auth.isAuthenticated()) {
+        userName = auth.getName();
+      } else {
+        LOG.info("Unauthenticated user in demo mode: current user will be default (if available)");
+      }
+      DemoUser found = config.getByUsername(userName);
+      if (found == null) {
+        LOG.error("Invalid user {} in demo mode", userName);
+      }
+      return Optional.ofNullable(found);
+    };
+  }
+
+  /**
+   * An {@link AuthorizationService} that looks up the username of the current authenticated user in
+   * the list of configured demo users, falling back to the default if one is configured, and
+   * returns the claims for that {@link DemoUser}. See {@link DemoUserIdentitySupplier} for snarky
+   * comments about this idea.
+   */
+  public static class DemoUserAuthorizationService implements AuthorizationService {
+
+    private final Supplier<Optional<DemoUser>> _getCurrentUser;
+
+    public DemoUserAuthorizationService(DemoUserConfiguration config) {
+      _getCurrentUser = getCurrentDemoUserSupplier(config);
+    }
+
+    public DemoUserAuthorizationService(List<DemoUser> users) {
+      this(new DemoUserConfiguration(users));
+    }
+
+    @Override
+    public List<OrganizationRoleClaims> findAllOrganizationRoles() {
+      return _getCurrentUser.get().map(DemoUser::getAuthorization).map(List::of).orElse(List.of());
```

Comment:
sigh, i feel like this is about to create a huge headache for me being unable to include facility UUID in my DemoUserConfiguration and thus having to change the config authorization type away from OrganizationRoleClaims

Reply:
also, this line has so much syntactic sugar in it that i think it needs a comment explaining what it's outputting

Reply:
actually, i take back comment #1 -- i realize that integrating Facility UUIDs into OrganizationRoleClaims creates the same conflict with the existing AuthorizationServiceConfig